app/script/components/participantAvatar.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.components = z.components || {};
+
+z.components.ParticipantAvatar = class ParticipantAvatar {
+  constructor(params, component_info) {
+    this.participant = ko.unwrap(params.participant);
+    this.isUser = this.participant instanceof z.entity.User && !this.participant.isBot;
+    this.isService = this.participant instanceof z.integration.ServiceEntity || this.participant.isBot;
+
+    this.delay = params.delay;
+    this.size = params.size;
+    this.element = $(component_info.element);
+
+    if (this.size) {
+      this.element.addClass(`${this.isService ? 'service' : 'user'}-avatar-${this.size}`);
+    }
+
+    this.avatarLoadingBlocked = false;
+    this.avatarEnteredViewport = false;
+
+    this.dispose = this.dispose.bind(this);
+
+    if (this.isUser) {
+      this.element.attr({
+        id: z.util.create_random_uuid(),
+        'user-id': this.participant.id,
+      });
+
+      this.initials = ko.pureComputed(() => {
+        if (this.element.hasClass('user-avatar-xs')) {
+          return z.util.StringUtil.get_first_character(this.participant.initials());
+        }
+        return this.participant.initials();
+      });
+
+      this.state = ko.pureComputed(() => {
+        switch (true) {
+          case this.participant.is_me:
+            return 'self';
+          case typeof params.selected === 'function' && params.selected():
+            return 'selected';
+          case this.participant.is_team_member():
+            return '';
+          case this.participant.is_blocked():
+            return 'blocked';
+          case this.participant.is_request():
+            return 'pending';
+          case this.participant.is_ignored():
+            return 'ignored';
+          case this.participant.is_canceled() || this.participant.is_unknown():
+            return 'unknown';
```

Comment:
If we are at it we can turn these into an enum

Reply:
They are just used here. So there is no typo hazard.